lib/bamboo/adapters/elastic_email_adapter.ex

```diff
@@ -197,10 +197,50 @@ defmodule Bamboo.ElasticEmailAdapter do
 
   defp put_transactional(map), do: Map.put(map, "isTransactional", true)
 
-  defp put_post_back(%{private: %{elastic_custom_vars: %{post_back: post_back}}} = map),
-    do: Map.put(map, "postBack", post_back)
+  defp put_elastic_send_options(%{private: %{elastic_custom_vars: options} = private} = email) do
+    private =
+      private
+      |> Map.delete(:elastic_custom_vars)
+      |> Map.put(:elastic_send_options, options)
 
-  defp put_post_back(map), do: map
+    put_elastic_send_options(%{email | private: private})
+  end
+
+  defp put_elastic_send_options(%{private: %{elastic_send_options: options}} = email) do
+    options
+    |> Enum.map(&send_option/1)
+    |> Enum.reject(&is_nil/1)
```

Comment:
Thinking about this, a small change:

```elixir
|> Enum.reject(&is_nil(&1) || match?({_, nil}, &1))
```

This uses `Kernel.match?/2` to make sure we’re not sending parameters that themselves are nil. (I don’t think that there’s any parameters where `param=` is really legitimate.)